test/nodes/test_retriever.py

```diff
@@ -650,6 +650,37 @@ def test_table_text_retriever_training(tmp_path, document_store, samples_path):
     )
 
 
+def test_sentence_transformers_retriever_training_with_gradient_checkpointing():
+    with patch("sentence_transformers.SentenceTransformer.fit") as mock_fit:
+        retriever = EmbeddingRetriever(
+            embedding_model="sentence-transformers/all-MiniLM-L6-v2",
+            model_format="sentence_transformers",
+            use_gpu=False,
+        )
+        retriever.train(
+            training_data=[
+                {
+                    "question": "What color is the sky?",
+                    "pos_doc": "The color of the sky is blue.",
+                    "neg_doc": "The color of the grass is brown.",
+                },
+                {
+                    "question": "What is the capital of Germany?",
+                    "pos_doc": "The capital of Germany is Berlin.",
+                    "neg_doc": "The capital of France is Paris.",
+                },
+            ],
+            train_loss="mnrl",
+            n_epochs=1,
+            batch_size=1,
+            num_warmup_steps=0,
+            gradient_checkpointing=True,
+        )
+        mock_fit.assert_called_once_with(
+            train_objectives=ANY, epochs=1, optimizer_params={"lr": 2e-5}, warmup_steps=0, use_amp=False
```

Comment:
@ZanSara I had to use the `ANY` from the `unittest.mock` library for the `train_objectives` because it does not seem possible to get the variable to match the one called within the training code. Specifically this variable includes a torch `DataLoader` which always seems to provide a different hash even when constructing it DataLoaders with the same parameters. After some experimentation I was unable to find out if it is possible to get two Dataloders to satisfy the `==` assertion so I went with `ANY` here.

Reply:
Alright, no problem!